File: official/docs/java/order/BuyOrder.java
import com.easypost.EasyPost;
import java.util.*;

public class BuyOrder {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");
        
        Order order = Order.retrieve("order_...");

        Map<String, Object> buyParams = new HashMap<String, Object>();
        buyParams.put("carrier", "FedEx");
        buyParams.put("service", "FEDEX_GROUND");

        order.buy(buyParams);

        System.out.println(order);
    }
}

File: official/docs/go/orders/buy_order.go
package example

import (
	"fmt"
	"os"

	"github.com/EasyPost/easypost-go/v2"
)

func main() {
	apiKey := os.Getenv("EASYPOST_API_KEY")
	client := easypost.New(apiKey)

	order, _ := client.BuyOrder("order_...", "FeEx", "FEDEX_GROUND")

	fmt.Println(order)
}
